src/lightning_fabric/loggers/logger.py

```diff
@@ -0,0 +1,136 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Abstract base class used to build new loggers."""
+
+from abc import ABC, abstractmethod
+from argparse import Namespace
+from functools import wraps
+from typing import Any, Callable, Dict, Optional, Union
+
+from torch import Tensor
+from torch.nn import Module
+
+from lightning_fabric.utilities.rank_zero import rank_zero_only
+
+
+class Logger(ABC):
+    """Base class for experiment loggers."""
+
+    @property
+    @abstractmethod
+    def name(self) -> Optional[str]:
+        """Return the experiment name."""
+
+    @property
+    @abstractmethod
+    def version(self) -> Optional[Union[int, str]]:
+        """Return the experiment version."""
+
+    @property
+    def root_dir(self) -> Optional[str]:
+        """Return the root directory where all versions of an experiment get saved, or `None` if the logger does
+        not save data locally."""
+        return None
+
+    @property
+    def log_dir(self) -> Optional[str]:
+        """Return directory the current version of the experiment gets saved, or `None` if the logger does not save
+        data locally."""
+        return None
+
+    @property
+    def group_separator(self) -> str:
+        """Return the default separator used by the logger to group the data into subfolders."""
+        return "/"
+
+    @abstractmethod
+    def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
+        """Records metrics. This method logs metrics as soon as it received them.
+
+        Args:
+            metrics: Dictionary with metric names as keys and measured quantities as values
+            step: Step number at which the metrics should be recorded
+        """
+        pass
```

Comment:
This should probably become `log_dict`, and another method `log`, to align with the rest Lightning's naming.

Reply:
I don't agree. What woudl be the difference between `log` and `log_dict` for this class specifically?

`log_metrics` is explicit and avoids collision with a polymorphic `log()` method if one of the implementations support it

Reply:
It depends on what product wants the logger interaction and experience to be. Will a user interact mainly with the fabric object and its log methods? Or will a user mainly interact with the logger object, i.e., `logger.log_()*`, `logger.experiment.xyz()`? 
Undoubtedly, power users will want to log media datatypes and not just metrics. I don't see how that would all be enabled through the fabric object. And then users would write code with a mix of `fabric.log` and `logger.log_metrics` and `logger.experiment.log_image`.

Reply:
Regardless of what `fabric.log()` supports, the logger interface don't need to have a matching name. This is how it is in PL at the moment.

Reply:
Not planning to change it in this PR. We should continue the discussion elsewhere, also in the context of what self.log should do and where in 2.0. Resolving